sentry-android-core/src/main/java/io/sentry/android/core/internal/gestures/SentryGestureListener.java

```diff
@@ -178,12 +192,131 @@ private void addBreadcrumb(
 
     hub.addBreadcrumb(
         Breadcrumb.userInteraction(
-            eventType, ViewUtils.getResourceId(target), className, additionalData),
+            eventType, ViewUtils.getResourceIdWithFallback(target), className, additionalData),
         hintMap);
   }
 
+  private void startTracing(final @NotNull View target, final @NotNull String eventType) {
+    if (!(options.isTracingEnabled() && options.isEnableUserInteractionTracing())) {
+      return;
+    }
+
+    final Activity activity = activityRef.get();
+    if (activity == null) {
+      options.getLogger().log(SentryLevel.DEBUG, "Activity is null, no transaction captured.");
+      return;
+    }
+
+    final String viewId;
+    try {
+      viewId = ViewUtils.getResourceId(target);
+    } catch (Resources.NotFoundException e) {
+      options
+          .getLogger()
+          .log(
+              SentryLevel.DEBUG,
+              "View id cannot be retrieved from Resources, no transaction captured.");
+      return;
+    }
+
+    final View view = activeView.get();
+    if (activeTransaction != null) {
+      if (target.equals(view)
+          && eventType.equals(activeEventType)
+          && !activeTransaction.isFinished()) {
+        options
+            .getLogger()
+            .log(
+                SentryLevel.DEBUG,
+                "The view with id: "
+                    + viewId
+                    + " already has an ongoing transaction assigned. Rescheduling finish");
+
+        final Long idleTimeout = options.getIdleTimeout();
+        if (idleTimeout != null) {
+          // reschedule the finish task for the idle transaction, so it keeps running for the same
+          // view
+          activeTransaction.scheduleFinish(idleTimeout);
+        }
+        return;
+      } else {
+        // as we allow a single UI transaction running on the bound Scope, we finish the previous
+        // one, if it's a new view
+        stopTracing(SpanStatus.OK);
+      }
+    }
+
+    // we can only bind to the scope if there's no running transaction
+    final String name = getActivityName(activity) + "." + viewId;
+    final String op = UI_ACTION + "." + eventType;
+    final ITransaction transaction =
+        hub.startTransaction(name, op, true, options.getIdleTimeout(), true);
```

Comment:
Since we have the instance of the `activity` here, we should consider adding support for slow and frozen frames as well.
https://github.com/getsentry/sentry-java/blob/main/sentry-android-core/src/main/java/io/sentry/android/core/ActivityLifecycleIntegration.java#L164-L175=
See the usage of `activityFramesTracker`.

Reply:
We haven't planned for that, but I'm fine doing it since it's a quick one, @philipphofmann @brustolin is it also feasible for you?

Reply:
Your call, file an issue otherwise, so we don't forget.

Reply:
We do this anyways for every transaction https://github.com/getsentry/sentry-cocoa/blob/e5a906bf07e0793d1d538ac33e2869efd0b668be/Sources/Sentry/SentryTracer.m#L99-L105

Reply:
ah I didn't know this, that's neat. I will do it as part of this PR then, np